presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PruneUnreferencedOutputs.java

```diff
@@ -460,6 +461,7 @@ public PlanNode visitMarkDistinct(MarkDistinctNode node, RewriteContext<Set<Symb
 
             ImmutableSet.Builder<Symbol> expectedInputs = ImmutableSet.<Symbol>builder()
                     .addAll(node.getDistinctSymbols())
+                    .addAll(node.getOrderBySymbols())
```

Comment:
I understand it might be confusing. I prefer the approach that when the change is introduced to an existing object, like adding new field in a plan node. Then I would love to enclose such change in one commit. That commit would also cover handling that new field properly in all the places where the given class is used so far according to this new field semantic. For example in places like `PruneUnreferencedOutputs` etc. Then in another commit I would use that field for the purposes it was design to, where you add new functionalities.

Though I understand that the border between these two things could seen differently.